src/Blocks/BlockIce.h

```diff
@@ -25,9 +25,47 @@ class cBlockIceHandler :
 		{
 			return cItem(m_BlockType);
 		}
-		else
+		return {};
+	}
+
+	virtual void OnUpdate(
+		cChunkInterface & a_ChunkInterface,
+		cWorldInterface & a_WorldInterface,
+		cBlockPluginInterface & a_PluginInterface,
+		cChunk & a_Chunk,
+		const Vector3i a_RelPos
+	) const override
+	{
+		// disappears instantly in nether
+		if (a_WorldInterface.GetDimension() == dimNether)
+		{
+			a_Chunk.SetBlock(a_RelPos, E_BLOCK_AIR, 0);
+			return;
+		}
+		// Artificial light on any of the surrounding block > 11 leads to melting the ice
+		const std::array<Vector3i, 7> Offsets =
+			{Vector3i(1, 0, 0), Vector3i(-1, 0, 0),
+			Vector3i(0, 1, 0), Vector3i(0, -1, 0),
+			Vector3i(0, 0, 1), Vector3i(0, 0, -1)};
+
+		for (const auto & Offset : Offsets)
 		{
-			return {};
+			auto Position = a_RelPos + Offset;
+			const auto Chunk = a_Chunk.GetRelNeighborChunk(Position.x, Position.z);
```

Comment:
You can GetRelNeighborChunkAdjustCoords to make Position relative to the returned chunk

Reply:
Oh so that's what that method does :D